src/renderer/routes/test_sequencer_panel/components/modals/CreatePlaceholderTestModal.tsx

```diff
@@ -0,0 +1,112 @@
+import { Button } from "@/renderer/components/ui/button";
+import { Dialog, DialogContent } from "@/renderer/components/ui/dialog";
+import { Input } from "@/renderer/components/ui/input";
+import {
+  createNewTest,
+  useDisplayedSequenceState,
+} from "@/renderer/hooks/useTestSequencerState";
+import { useState } from "react";
+import { toast } from "sonner";
+
+export const CreatePlaceholderTestModal = ({
+  isModalOpen,
+  setModalOpen,
+}: {
+  isModalOpen: boolean;
+  setModalOpen: (value: boolean) => void;
+}) => {
+  const [name, setName] = useState<string>("");
+  const [min, setMin] = useState<number>(0);
+  const [max, setMax] = useState<number>(0);
+  const [unit, setUnit] = useState("");
+
+  const { addNewElems } = useDisplayedSequenceState();
+
+  const handleCreate = async () => {
+    if (name === "") {
+      toast.error("Please enter a test name");
+    }
+    const res = await addNewElems([
+      createNewTest(
+        name,
+        "",
+        "placeholder",
+        false,
+        undefined,
+        undefined,
+        min,
+        max,
+        unit,
+      ),
+    ]);
+    if (res.isErr()) {
+      return;
+    }
+    setModalOpen(false);
+  };
+
+  return (
+    <Dialog open={isModalOpen} onOpenChange={setModalOpen}>
+      <DialogContent>
+        <h2 className="text-lg font-bold text-accent1">
+          Create a placeholder test step
+        </h2>
+        <p className="text-xs text-muted-foreground">
+          {" "}
+          This will create a new test step with the given name and expected
+          value without being link to any code. The code can be added later.
+        </p>
+
+        <div className="grow pb-1 text-xs">
```

Comment:
I would actually recommend using a form to handle those input states, which will also let you do validation easily.